l10n_ar_ux/models/account_move.py

```diff
@@ -124,3 +125,13 @@ def _l10n_ar_get_document_number_parts(self, document_number, document_type_code
         # forzar unicidad con cambios de approach al ir migrando de versiones
         document_number = document_number.split('(')[0]
         return super()._l10n_ar_get_document_number_parts(document_number, document_type_code)
+
+    def button_cancel(self):
+        # Evitamos que se pueda cancelar una factura que ya fue previamente confirmada y enviada a AFIP.
+        # Este caso se da cuando dos usuarios están a la vez editando la misma factura, uno confirma
+        # y el otro, sin refrescar, cancela.
+        for record in self:
+            if record.state == "posted" and record.invoice_filter_type_domain == "sale" \
+                    and record.l10n_ar_afip_auth_mode == "CAE" and record.l10n_ar_afip_auth_code:
+                raise UserError(_("You cannot cancel a posted document."))
+        return super().button_cancel()
```

Comment:
1. Instead of a for loop you can do a filtered() and show the info of all the invoices that fulfill the condition.
2. Extend the message to ensure the info that is related to AFIP-posted documents
3. Since this is made in 16.0 translations will not be sync :(. please create the translation manually or just put the message in Spanish (in the FW to 18.0 you can change the message to English).
4. if you don't like the filtered and prefer the for loop please use move instead of record as the loop variable name,


```suggestion
        if posted_in_afip :=  self.filtered(lambda x: x..state == "posted" and x.invoice_filter_type_domain == "sale" 
           and x.l10n_ar_afip_auth_mode == "CAE" and x.l10n_ar_afip_auth_code):
                raise UserError(_("You cannot cancel documents already posted in AFIP (%s).", ",".join(posted_in_afip.mapped('name'))))
        return super().button_cancel()
```